File: examples/simple-asset-transfer/corda/clients/src/main/kotlin/com/accenture/interoperability/webserver/AuthController.kt
package com.accenture.interoperability.webserver

import org.springframework.beans.factory.annotation.Value
import org.springframework.http.HttpStatus
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Component
import org.springframework.web.bind.annotation.*


/**
 * Authentication configuration.
 */
@Component
class SingleUser(
    @Value("\${auth.username}") private val username: String,
    @Value("\${auth.password}") private val password: String
) {
    internal fun isValidSingleUser(
        username: String, password: String
    ) = this.password == password && this.username == username
}

/**
 * Authentication API endpoints.
 */
@RestController
@RequestMapping("/api/v1/auth")
class AuthController(
    private val singleUser: SingleUser,
    private val jwtConfig: JwtConfig
) {

    class LoginRequestBody(
        val username: String,
        val password: String
    )

    /**
     * Endpoint for user authentication.
     */
    @PostMapping("/login", produces = [MediaType.APPLICATION_JSON_VALUE])
    fun postLogin(
        @RequestBody requestBody: LoginRequestBody
    ) = if (singleUser.isValidSingleUser(requestBody.username, requestBody.password))
        ResponseEntity.ok(
            mapOf("token" to jwtConfig.jwtToken(requestBody.username))
        )
    else
        ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("Authentication error")


    @GetMapping("/random-number", produces = [MediaType.APPLICATION_JSON_VALUE])
    fun getRandomNumber() = ResponseEntity.ok(
        mapOf("num" to (Math.random() * 100).toInt())
    )
}

File: contribs/Accenture-BIF/core/examples/simple-asset-transfer/besu/api/routes/auth/auth.controller.js
const httpStatus = require('http-status');
const JwtToken = require('../../utils/JwtToken');
const APIError = require('../../utils/APIError');
const config = require('../../config/config');

/**
 * Returns jwt token if valid username and password is provided
 * @property {string} req.body.username - The username of user.
 * @property {string} req.body.password - The password of user.
 * @returns {*}
 */
async function login(req, res, next) {
  const { username, password } = req.body;

  const isUserEqual = username === config.user.username && password === config.user.password;
  if (isUserEqual) {
    const token = JwtToken.sign(config.user);
    return res.json({ token });
  }

  // Authentication error handler
  const err = new APIError('Authentication error', httpStatus.UNAUTHORIZED, true);
  return next(err);
}

/**
 * This is a protected route. Will return random number only if jwt token is provided in header.
 * @param req
 * @param res
 * @returns {*}
 */
function getRandomNumber(req, res) {
  // req.user is assigned by jwt middleware if valid token is provided
  return res.json({
    user: req.user,
    num: Math.random() * 100,
  });
}

module.exports = { login, getRandomNumber };
